### PR TITLE
Add single static instance of SpecialPermission

### PR DESCRIPTION
**Pull request by Tim-Brooks**
This commit adds a SpecialPermission constant and uses that constant
opposed to introducing new instances everywhere.

Additionally, this commit introduces a single static method to check that
the current code has permission. This avoids all the duplicated access
blocks that exist currently.

### CODE REVIEW

**Review by jasontedor — APPROVED** [review]
LGTM. I'm good with it, let's make sure @s1monw is before this is merged in.

File: modules/lang-expression/src/main/java/org/elasticsearch/script/expression/ExpressionScriptEngineService.java [hunk 1]
```diff
@@ -79,7 +79,7 @@ public Object compile(String scriptName, String scriptSource, Map<String, String
         // classloader created here
         final SecurityManager sm = System.getSecurityManager();
         if (sm != null) {
-            sm.checkPermission(new SpecialPermission());
+            sm.checkPermission(SpecialPermission.INSTANCE);
```
Comment (rjernst) [inline modules/lang-expression/src/main/java/org/elasticsearch/script/expression/ExpressionScriptEngineService.java]: Can this be replaced with `SpecialPermission.checkSpecialPermission()`?

File: core/src/main/java/org/elasticsearch/SpecialPermission.java [hunk 2]
```diff
@@ -76,4 +79,14 @@ public SpecialPermission() {
     public SpecialPermission(String name, String actions) {
         this();
     }
+
+    /**
+     * Check that the current stack has {@link SpecialPermission} access according to the {@link SecurityManager}.
+     */
+    public static void checkSpecialPermission() {
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/SpecialPermission.java]: Maybe just call this method `check()` since it is used statically?

**Review by rjernst — APPROVED** [review]
LGTM too

File: modules/lang-expression/src/main/java/org/elasticsearch/script/expression/ExpressionScriptEngineService.java (hunk 1 above)
Comment (tbrooks8) [inline modules/lang-expression/src/main/java/org/elasticsearch/script/expression/ExpressionScriptEngineService.java]: Yes. It is slightly different in form which is why autorefactoring probably did not catch it.

File: core/src/main/java/org/elasticsearch/SpecialPermission.java (hunk 2 above)
Comment (tbrooks8) [inline core/src/main/java/org/elasticsearch/SpecialPermission.java]: Yeah. I added a committing changing that.

**Review by s1monw — APPROVED** [review]
LGTM 2

**Pull request merged**